pkg/pulumiyaml/run.go

```diff
@@ -133,6 +133,10 @@ func LoadDir(directory string) (*ast.TemplateDecl, syntax.Diagnostics, error) {
 		return nil, diags, err
 	}
 
+	if diags.HasErrors() {
```

Comment:
Shouldn't we still do SeachPackageDecls so we can add that to the diagnostics as well if it errors but then check if template is nil just before assigning Packages to it?

Reply:
I mean if the template has errors while parsing, we would early exit like most Go functions, unless it is a pattern to combine errors from functions downstream. I don't have a strong opinion about this, could fix it either way

Reply:
Yeh fine, easy to change if a compelling reason comes up later